etsin_finder/frontend/js/components/dataset/index.jsx

```diff
@@ -89,32 +88,26 @@ class Dataset extends React.Component {
   updateData(isLive) {
     const researchDataset = this.state.dataset.research_dataset
 
-    const title = researchDataset.title;
     const description = researchDataset.description.map(single => (
       checkDataLang(single)
     ));
-    this.setState({ title, description });
-
-    if (isLive) {
-      const {
-        creator,
-        contributor,
-        issued,
-        rights_holder,
-      } = this.state.dataset.research_dataset;
-      this.setState({
-        creator,
-        contributor,
-        issued,
-        rights_holder,
-      })
-    } else {
-      const {
-        urn_identifier,
-        preferred_identifier
-      } = researchDataset;
-      this.setState({ urn_identifier, preferred_identifier, date_modified: this.state.dataset.date_modified })
-    }
+
+    const {
+      title,
+      creator,
+      contributor,
+      issued,
+      rights_holder
+    } = this.state.dataset.research_dataset;
+    this.setState({
+      title,
+      description,
+      creator,
+      contributor,
+      issued,
+      rights_holder
+    })
+
     this.setState({ live: isLive });
     this.setState({ loaded: true })
```

Comment:
Nää kaikki state muutokset vois yhdistää yhdeksi. 3 -> 1.
`this.setState({
title,
description,
creator,
contributor,
issued,
rights_holder,
live: isLive,
loaded: true
)}`